### PR TITLE
[CSLD58] Swagger wallet API docs online

### PR DESCRIPTION
**Pull request by denisshevchenko**

### CODE REVIEW

File: src/wallet-web-api-swagger/Main.hs [hunk 1]
```diff
@@ -1,43 +1,39 @@
...
 {-# LANGUAGE KindSignatures      #-}
-{-# LANGUAGE OverloadedStrings   #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleInstances   #-}
+{-# LANGUAGE OverloadedStrings   #-}
 
 -- | This program builds Swagger specification for wallet web API and converts it to JSON.
 -- We run this program during CI build.
--- Produced JSON will be used to create online
+-- Produced JSON will be used to create online
```
Comment (neongreen) [inline src/wallet-web-api-swagger/Main.hs]: Trailing whitespace

File: src/wallet-web-api-swagger/Main.hs [hunk 2]
```diff
@@ -1,43 +1,39 @@
-{-# LANGUAGE FlexibleInstances   #-}
 {-# LANGUAGE KindSignatures      #-}
-{-# LANGUAGE OverloadedStrings   #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleInstances   #-}
```
Comment (neongreen) [inline src/wallet-web-api-swagger/Main.hs]: I feel that stylish-haskell wasn't applied to this file – can you apply it? (with standard Serokell settings)

File: src/wallet-web-api-swagger/Main.hs [hunk 3]
```diff
@@ -83,7 +79,10 @@ instance ToSchema      BackupPhrase
 -- We need this instance for correct Swagger-specification.
 instance {-# OVERLAPPING #-} (Typeable a, ToSchema a) => ToSchema (Either WalletError a) where
     declareNamedSchema proxy = genericDeclareNamedSchema defaultSchemaOptions proxy
-        & mapped . name ?~ toText ("Either WalletError " ++ show (typeOf (undefined :: a)))
+        & mapped . name ?~ toText ("Either WalletError " ++ show (typeRep right))
```
Comment (neongreen) [inline src/wallet-web-api-swagger/Main.hs]:
``` haskell
mapped . name ?~ show (typeRep (Proxy @(Either WalletError a)))
```

Or at the very least don't use `toText`, it's not needed here. `show` can produce `Text`.

**Comment by neongreen** [comment]
> continuous-integration/travis-ci/push
> The job exceeded the maximum time limit for jobs, and has been terminated.

Darn. @domenkozar can we do anything about it? (Also, what's the difference between `pr` and `push`?)

**Comment by gromakovsky** [comment]
I approved, but I am afraid of merging it, because of CI failure (timeout). AFAIU `update_wallet_web_api_docs.sh` took only 63.71 sec http://i.imgur.com/WdJWPIl.png so I think it shouldn't be a problem. It's strange. Let's wait for @domenkozar's suggestions about it.

**Comment by denisshevchenko** [comment]
I think we _can_ merge it. I just restarted this build - and it's ok now. Dunno why build could take so long time...

**Comment by domenkozar** [comment]
Looks good. The build takes 1h17min, our limit is now 1h20min so we will see more timeouts.

We keep adding more things to the build process, it's best if we just ask travis-ci to support our needs.

On the long run we will be transitioning to Nix-only infrastructure that will make life easier.

**Pull request merged**